### PR TITLE
Migrate pallet-preimage to benchmark v2

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Part of:

- #6202.

### CODE REVIEW

File: substrate/frame/preimage/src/benchmarking.rs [hunk 1]
```diff
@@ -43,206 +42,225 @@ fn sized_preimage_and_hash<T: Config>(size: u32) -> (Vec<u8>, T::Hash) {
...
+
+		assert!(!Pallet::<T>::have_preimage(&hash));
 	}
 
 	// Expensive request - will unreserve the noter's deposit.
-	request_preimage {
+	#[benchmark]
+	fn request_preimage() {
+		let o = T::ManagerOrigin::try_successful_origin()
+			.expect("ManagerOrigin has no successful origin required for the benchmark");
```
Comment (ggwpez) [inline substrate/frame/preimage/src/benchmarking.rs:139]: Why is this an expect now? The pallet is also usable without manager origin.  
It returned `BenchmarkError::Weightless` before.

Comment (AurevoirXavier) [reply substrate/frame/preimage/src/benchmarking.rs:139]: I think that error is meaningless.

Same as the https://github.com/paritytech/polkadot-sdk/blob/06debd0b408cc4308079d9f980245939697fdfa8/substrate/frame/treasury/src/benchmarking.rs#L121-L126.

It was not part of the benchmark code block. Additionally, I believe the original `.expect` message is more readable. So I reuse it.

What do you suggest?

Comment (AurevoirXavier) [reply substrate/frame/preimage/src/benchmarking.rs:139]: Also can be found here

https://github.com/paritytech/polkadot-sdk/blob/06debd0b408cc4308079d9f980245939697fdfa8/substrate/frame/lottery/src/benchmarking.rs#L50

**Pull request merged**